ReactNativeClient/lib/joplin-database.js

```diff
@@ -873,13 +883,36 @@ class JoplinDatabase extends Database {
 		return true;
 	}
 
+	async fuzzySearchEnabled() {
+		try {
+			await this.selectOne('SELECT count(*) FROM notes_spellfix');
+		} catch (error) {
+			this.logger().warn('Fuzzy search check failed', error);
+			return false;
+		}
+
+		this.logger().info('Fuzzy search check succeeded');
+
+		return true;
+	}
+
+
 	version() {
 		return this.version_;
 	}
 
 	async initialize() {
 		this.logger().info('Checking for database schema update...');
 
+		try {
+			console.log('Trying to load spellfix');
+			console.log(`Current directory: ${process.cwd()}`);
+			await this.loadExtension('./lib/sql-extensions/spellfix');
+			console.log('Spellfix extension loaded!');
```

Comment:
This relative path only works for ElectronClient.

When running CliClient the current working directory would be `CliClient/` while the lib folder is inside `CliClient/build`
Is there a better way to do this?

Reply:
Is it not possible to install it via npm? Because we also need to make sure it's compatible with whatever version of the Sqlite lib is installed via npm.

Reply:
No. Spellfix is a loadable extension. Its source code can be found here: https://www.sqlite.org/src/dir?ci=tip&name=ext/misc

From the documentation, we need to compile the C source code to different platforms (unix, windows) like this: https://www.sqlite.org/loadext.html#compiling_a_loadable_extension

Since it is an official extension wouldn't compatibility be guaranteed by sqlite?